sdks/java/io/elasticsearch/src/main/java/org/apache/beam/sdk/io/elasticsearch/ElasticsearchIO.java

```diff
@@ -2854,9 +2865,28 @@ private List<Document> flushBatch() throws IOException, InterruptedException {
           responseEntity = handleRetry("POST", endPoint, Collections.emptyMap(), requestBody);
         }
 
-        List<Document> responses =
-            createWriteReport(
-                responseEntity, spec.getAllowedResponseErrors(), spec.getThrowWriteErrors());
+        List<Document> responses;
+        // If org.elasticsearch.client.ResponseException was thrown, return all input Documents with
+        // withHasError(true)
+        // so that they could be caught by FAILED_WRITES tag.
+        if (elasticResponseExceptionMessage != null) {
+          String errorJsonMessage =
+              String.format(
+                  "{\"message\":\"org.elasticsearch.client.ResponseException was thrown in batch flush: %s\"}",
+                  elasticResponseExceptionMessage);
+          responses =
+              inputEntries.stream()
+                  .map(
+                      doc ->
+                          Document.create()
```

Comment:
I believe this should be `doc -> doc.withHasError(true) ...` rather than `doc -> Document.create()` in order to preserve the other information contained in the failed documents.